paramiko/ecdsakey.py

```diff
@@ -160,9 +160,9 @@ def __init__(
 
             pointinfo = msg.get_binary()
             try:
-                numbers = ec.EllipticCurvePublicNumbers.from_encoded_point(
+                numbers = ec.EllipticCurvePublicKey.from_encoded_point(
                     self.ecdsa_curve.curve_class(), pointinfo
-                )
+                ).public_numbers()
```

Comment:
You can just leave this as a key, the next line is converting the numbers to a key :-)